src/main/java/Friend.java

```diff
@@ -81,13 +69,19 @@ public boolean equals(Object o) {
         if (o == null) {
             return false;
         }
-        // if o and this are of different classes, they're not the same
+        // if o and this are of different classes, they're not the equal
         if (o.getClass() != this.getClass()) {
             return false;
         }
         // Cast o as a friend
         Friend other = (Friend) o;
-        return this.firstName.equals(other.firstName)
-                && this.lastName.equals(other.lastName);
+        return this.firstName.equals(other.firstName) &&
+                this.lastName.equals(other.lastName) &&
+                this.email.equals(other.email);
```

Comment:
It's typical to prefer `Objects.equals(this.firstName, that.firstName) && ...` for the null checks it does.